docarray/array/mixins/io/pushpull.py

```diff
@@ -126,21 +159,34 @@ def pull(
         """Pulling a :class:`DocumentArray` from Jina Cloud Service to local.
 
         :param token: the upload token set during :meth:`.push`
+        :param id: the unique identifier that returned from :meth:`.push`
         :param show_progress: if to show a progress bar on pulling
         :param local_cache: store the downloaded DocumentArray to local folder
         :return: a :class:`DocumentArray` object
         """
 
         import requests
 
-        url = f'{_get_cloud_api()}/v2/rpc/da.pull?token={token}'
-        response = requests.get(url)
+        headers = {}
+
+        auth_token = _get_hub_config().get('auth_token')
```

Comment:
so it's either you provide an `auth_token` if you're logged in or a DA `token` if you're not ?

Reply:
You can use DA `token` even when you're logged in.
User login status and DA `id`/`token` params are not related.
You can pull other users' DA by `id` if it is public.
API has an ability to set visibility (`public` or `private`) but is not added to this mixin yet.